packages/apps/virtual-machine/templates/secret.yaml

```
@@ -9,13 +9,21 @@ stringData:
 metadata:
   name: {{ include "virtual-machine.fullname" . }}-cloud-init
 stringData:
   userdata: |
-    {{- .Values.cloudInit | nindent 4 }}
+    {{- if .Values.cloudInit }}
+    {{-   .Values.cloudInit | nindent 4 }}
+    {{- else if and (.Values.sshKeys) (not .Values.cloudInit) }}
+    #cloud-config
+    ssh_authorized_keys:
+      {{- range .Values.sshKeys }}
+      - {{ quote $v }} 
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)